drop-transfer/src/ws/server/v4.rs

```diff
@@ -515,7 +515,12 @@ impl Downloader {
 
 #[async_trait::async_trait]
 impl handler::Downloader for Downloader {
-    async fn init(&mut self, task: &super::FileXferTask) -> crate::Result<handler::DownloadInit> {
+    async fn init(
+        &mut self,
+        task: &super::FileXferTask,
+        events: Arc<FileEventTx<IncomingTransfer>>,
```

Comment:
I'm 99% sure this can be done by passing `events` by reference

Reply:
It emits
```
`events` does not live long enough
borrowed value does not live long enough
```

Reply:
in the `start_download()` in the end if I pass in a reference here. This is due to the fact that the events is passed down from elsewhere and it comes in as Arc.